packages/rx-effects-react/src/useObserver.ts

```diff
@@ -12,32 +12,81 @@ import { useConst } from './useConst';
  *
  * @example
  * ```ts
- * const observer = useCallback((nextValue) => {
+ * useObserver(source$, (nextValue) => {
  *   logger.log(nextValue);
- * }, []);
- * useObserver(source$, observer);
+ * });
  * ```
  */
 export function useObserver<T>(
   source$: Observable<T>,
-  observerOrNext: Partial<Observer<T>> | ((value: T) => void),
+  observerOrNext: PartialObserver<T> | ((value: T) => void),
 ): Subscription {
   const hookSubscriptions = useConst(() => new Subscription());
 
+  const argsRef = useRef<PartialObserver<T> | ((value: T) => void)>(
+    observerOrNext,
+  );
+
+  // Update the latest observable and callbacks
+  // synchronously after render being committed
+  useIsomorphicLayoutEffect(() => {
+    argsRef.current = observerOrNext;
```

Comment:
Let's store an observer object in the ref:

```suggestion
    argsRef.current = typeof observerOrNext === 'function' ? {next: observerOrNext} : observerOrNext;
```

It helps to remove `typeof` checks from the observe proxy.

Reply:
I like it, but second variant is better for performance

first variant
```js
  const argsRef = useRef<(value: T) => void>(
    typeof observerOrNext === 'function' ? {next: observerOrNext} : observerOrNext, // will be executed on each render
  );

  useIsomorphicLayoutEffect(() => {
    argsRef.current = typeof observerOrNext === 'function' ? {next: observerOrNext} : observerOrNext; // will be executed on each render
  });

```

second variant
```js
    const subscription = source$.subscribe({
      next: (value) => {
        const nextObserver =
          typeof argsRef.current === 'function'
            ? argsRef.current
            : argsRef.current.next;  // will be executed on each event

        if (nextObserver) {
          return nextObserver(value);
        }
      },
```

Reply:
Actually, `undefined` can be used as the initial value for the ref:

```ts
const observerRef = useRef<PartialObserver<T> | undefined>();

  // Update the latest observable and callbacks
  // synchronously after render being committed
  useIsomorphicLayoutEffect(() => {
    observerRef.current =
      typeof observerOrNext === 'function'
        ? { next: observerOrNext }
        : observerOrNext;
  });

  useEffect(() => {
    if (hookSubscriptions.closed) {
      return;
    }

    const subscription = source$.subscribe({
      next: (value) => observerRef.current?.next?.(value),
      error: (error) => observerRef.current?.error?.(error),
      complete: () => observerRef.current?.complete?.(),
    });

    hookSubscriptions.add(subscription);
    
    return () => subscription.unsubscribe();
  }, [hookSubscriptions, source$]);
```

Reply:
What do you think?

Reply:
i think it's good.

i added `useRef<Partial<Observer<T>>>({});` to the new commit.
Now the optional chaining operator is not needed.
before `next: (value) => observerRef.current?.next?.(value),`
after `next: (value) => observerRef.current.next?.(value),`